packages/form-upload/Upload.js

```diff
@@ -84,9 +87,15 @@ const Upload = ({
           );
         }
 
-        await Promise.all(uploadResults);
-      } catch {
+        const responses = await Promise.all(uploadResults);
+        if (typeof onDeliverySuccess === 'function') {
+          onDeliverySuccess(responses);
```

Comment:
I'm wondering if we should push the `onDeliverySuccess` and `onDeliveryError` callbacks onto the uploads themselves... This would keep it more in line with our example for `onFileUpload` callbacks, and then the user's function doesn't need to be concerned with looping through the response data. I also think this would take advantage of what we have in `upload-core` where each callback pushed to an upload's `onSuccess` or `onError` array gets called for us. What do you think?

https://github.com/Availity/sdk-js/blob/master/packages/upload-core/src/upload.js#L136

https://github.com/Availity/sdk-js/blob/master/packages/upload-core/src/upload.js#L402-L411

```js
const Example = () => {
  const onUpload = (upload) => {
    if (upload) {
      upload.onSuccess.push(async () => {
        // success action
      });
      upload.onError.push(() => {
        // error action
      });
    }
  };

  const onRemove = (file) => {
    // remove action
  };

  return (
    <Form initialValues={{ myFile: undefined }}>
      <Upload
        name="myFile"
        btnText="Upload a claim"
        clientId="a"
        bucketId="b"
        customerId="c"
        onFileUpload={onUpload} // <-- add file callback function
        onFileRemove={onRemove} // <-- remove file callback function
        max={1}
      />
    </Form>
  );
};
```

Reply:
Looked at this again, overlooked the fact that `upload.onSuccess` and `upload.onError` are for executing callbacks after uploading to `vault`. So we would either need to pass the results of the file delivery call to the next callback, or keep everything as-is